CDP4Dal/Session.cs

```diff
@@ -686,12 +691,26 @@ public async Task Reload()
             }
         }
 
+        /// <summary>
+        /// Can a Cancel action be executed?
+        /// </summary>
+        /// <returns>True is Cancel is allowed, otherwise false.</returns>
+        public bool CanCancel()
+        {
+            if (this.cancellationTokenSource == null)

```

Comment:
the session object is multi-threaded... so, this may become a problem i think. Which read/write is being cancelled?